cmd/talosctl/cmd/talos/root.go

```diff
@@ -35,16 +36,20 @@ var GlobalArgs global.Args
 
 const pathAutoCompleteLimit = 500
 
+var defaultDialOpts = []grpc.DialOption{
+	grpc.WithContextDialer(dialer.DynamicProxyDialer),
```

Comment:
I wonder if we should inject that in the client itself (?) or is it going to be too much?

Reply:
This would require breaking change in `pkg` (which is public), because (IMHO) the best solution is to move the dialer package to `pkg/machinery` which is separate module.